src/provider/bn256_grumpkin.rs

```diff
@@ -35,6 +33,32 @@ pub mod grumpkin {
   };
 }
 
+#[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
+fn bn256_msm(points: &[bn256::Affine], scalars: &[bn256::Scalar]) -> bn256::Point {
+  cfg_if::cfg_if! {
+    if #[cfg(feature = "cuda")] {
+      let stream = ingonyama_grumpkin_msm::Config::new();
+      let cfg = ingonyama_grumpkin_msm::default_config(&stream.stream);
```

Comment:
Could we make this a `static` to not create it on every call?

Reply:
I tried this, but ran into issues like
```
error[E0277]: `*mut icicle_cuda_runtime::bindings::CUstream_st` cannot be sent between threads safely
   --> src/provider/bn256_grumpkin.rs:39:16
    |
39  | static STREAM: Lazy<ingonyama_grumpkin_msm::Config> = Lazy::new(|| ingonyama_grumpkin_msm::Config::new());
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `*mut icicle_cuda_runtime::bindings::CUstream_st` cannot be sent between threads safely
    |
    = help: within `ingonyama_grumpkin_msm::Config`, the trait `Send` is not implemented for `*mut icicle_cuda_runtime::bindings::CUstream_st`
```
Dmytro [mentioned](https://folding-implementers.zulipchat.com/#narrow/dm/639273,651996,670525,693329-group/near/426492184) that the first stream creation will have some delay, but each time after that, the Icicle/CUDA runtime will be able to reuse them with little overhead. 
> Yeah, unfortunately the first CUDA call in any application always incurs some latency, it's one-time and fairly constant, 70 ms. in your case. If you try to do the same operations (create stream, alloc memory) again, the time will no longer be noticeable.